tiatoolbox/visualization/tileserver_api.yml

```diff
@@ -0,0 +1,290 @@
+      parameters:
+        - in: path
+          name: prop
+          description: Property to update
+          required: true
+          schema:
+            type: string
+        - in: path
+          name: val
+          description: New value for the property
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful response
+
+  /tileserver/change_secondary_cmap/{type_id}/{prop}/{cmap}:
+    put:
+      summary: Change secondary color map for a property type
+      parameters:
+        - in: path
+          name: type_id
+          description: The type for which you want to override the primary
+            color map
+          required: true
+          schema:
+            type: string
+        - in: path
+          name: prop
+          description: Property to use to color annotations of the chose type
+          required: true
+          schema:
+            type: string
+        - in: path
+          name: cmap
+          description: Color map to use for the property
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful response
+
+  /tileserver/get_prop_names/{ann_type}:
+    get:
+      summary: Get all the properties of the annotations in the store
+      description: This endpoint retrieves all the property names in the annotations in the store of the specified type.
+      parameters:
+        - name: ann_type
+          in: path
+          required: true
+          description: The annotation type to filter properties by, or "all" for all types.
+          schema:
+            type: string
+      responses:
+        '200':
+          description: List of annotation properties.
+          content:
+            application/json:
+              schema:
+                type: array
+
+  /tileserver/get_prop_values/{prop}/{ann_type}:
+    get:
+      summary: Get a list of available values for a property
+      parameters:
+        - in: path
+          name: prop
+          description: Name of the property to retrieve values for
+          required: true
+          schema:
+            type: string
+        - in: path
+          name: type
+          description: Name of a type to retrieve property values for.
+            if all, prop values over all annotations will be returned
+          required: false
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  values:
+                    type: array
+                    items:
+                      type: string
+
+  /tileserver/reset:
+    get:
+      summary: Remove no longer needed objects if user closed the browser
+      responses:
+        '200':
+          description: Successful response
```

Comment:
Is this action on some kind of timeout? E.g. if the user loses connection or the browser crashes? Else the clean up will never be called.

Reply:
Bokeh has some hooks you can use to do stuff when specific things happen. So in the bokeh app, I have this set up to be called when a bokeh session is destroyed (eg if they close the window)